llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9430,7 +9552,21 @@ void VPWidenMemoryInstructionRecipe::execute(VPTransformState &State) {
   State.setDebugLocFrom(LI->getDebugLoc());
   for (unsigned Part = 0; Part < State.UF; ++Part) {
     Value *NewLI;
-    if (CreateGatherScatter) {
+    if (State.EVL) {
+      assert(State.UF == 1 && "Expected only UF == 1 when vectorizing with "
+                              "explicit vector length.");
+      Value *EVL = State.get(State.EVL, VPIteration(0, 0));
```

Comment:
Would be good to assert that State.EVL is an EVL VPInstruction here and above, unless that's done elsewhere already

Reply:
Already done.